File: app/src/main/java/com/privat/contacts/di/modules/AppModule.java
package com.privat.contacts.di.modules;

import android.content.Context;

import androidx.annotation.NonNull;

import com.privat.contacts.ContactsApplication;

import dagger.Module;
import dagger.Provides;

@Module
public class AppModule {
    @Provides
    Context providesContext(@NonNull ContactsApplication application) {
        return application.getApplicationContext();
    }
}

File: app/src/main/java/com/privat/contacts/di/modules/AppModule.kt
package com.privat.contacts.di.modules

import android.content.Context
import com.privat.contacts.ContactsApplication
import dagger.Module
import dagger.Provides

@Module
class AppModule {
    @Provides
    fun providesContext(application: ContactsApplication): Context {
        return application.applicationContext
    }
}
